### PR TITLE
Add nuget release version suffix to support publishing rcs to nuget.org

### PR DESCRIPTION
**Pull request by natke**
Re-create changes from a branch instead of a fork so that they can be tested in ADO.

Supersedes #15764

### CODE REVIEW

**Comment by natke** [comment]
Have tested with -alpha suffix.

**Comment by natke** [comment]
This is ready for review now.

I have manually tested it with the following combinations:
Nightly build config: - IsReleaseBuild = False, PreReleaseVersionSuffixString != none (just in case), PreReleaseVersionSuffixNumber != 0
RC build config: - IsReleaseBuild = True, PreReleaseVersionSuffixString = -alpha, PreReleaseVersionSuffixNumber = 1
Release build config: - IsReleaseBuild = True, PreReleaseVersionSuffixString = none, PreReleaseVersionSuffixNumber = 0

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 1]
```diff
@@ -15,15 +15,30 @@ parameters:
...
+  - -beta
+  - -rc
+  - none
+  default: none
+
+- name: PreReleaseVersionSuffixNumber
+  displayName: Number added to pre-release package version. Only used if IsReleaseBuild is true. Denotes the sequence of a pre-release package.
+  type: number
+  default: 0
+
```
Comment (skottmckay) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:41]: This seems potentially error prone. How does someone running this discover what is/isn't allowed in the suffix and what the last value/s used where?

Is there a reason not to use a timestamp and commit id as the suffix like we do with the nightly packages? That's guaranteed to result in the latest package being used, provides important information on what's in the package, and will be guaranteed to sort correctly when nuget looks to see if there are updated packages.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:41]: ] Azure Devops's web ui will show the "PreReleaseVersionSuffixString" parameter as a drop down box. So it's easy to use. It won't let you type strings there.

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml (hunk 1 above)
Comment (natke) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:41]: Not having the date time string was the requested developer experience by the user. Open to suggestions to make it less error prone. The generation is manual, as is the nuget upload (for better or worse), so whoever is running the pipeline should be aware of the required number.

**Pull request merged**